pic-sure-hpds-ui/src/main/webapp/picsureui/dataset/dataset-manage.js

```diff
@@ -0,0 +1,216 @@
+                        style: "action",
+                        handler: this.archiveRow(false)
+                    }
+                ]
+            }[tableId];
+
+            const columns = [ 
+                ...this.staticColumns,
+                ...actions.map(({title}) => ({title}))
+            ];
+
+            const columnDefs = [
+                { // format uquery created as date
+                    targets: 2,
+                    render: seconds => new Date(seconds).toDateString()
+                },
+                { // hide query id column
+                    targets: [3, 4],
+                    visible: false
+                },
+                ...actions.map(({name, style}, index) => ({
+                    data: null,
+                    render: row => this.renderButton(row.uuid, name, style),
+                    targets: this.staticColumns.length + index
+                }))
+            ];
+
+            const handlers = actions.map(({ name, handler }) => ({ name, handler }));
+
+            return { columns, columnDefs, handlers };
+        },
+        loadDatasets: function(){
+            if (overrides.loadDatasets){
+                overrides.loadDatasets(this);
+                return;
+            }
+
+            spinner.medium(
+                $.ajax({
+                    url: window.location.origin + "/picsure/dataset/named",
+                    type: "GET",
+                    headers: { "Authorization": "Bearer " + JSON.parse(sessionStorage.getItem("session")).token },
+                    contentType: "application/json",
+                    dataType: "text",
+                    success: function (response) {
+                        const datasets = JSON.parse(response)
+                        const { false: active = [], true: archived = [] } = _.groupBy(datasets, (dataset) => dataset.archived);
+    
+                        this.renderTable("active", active);
+                        this.renderTable("archived", archived);
+                    }.bind(this),
+                    error: function (response, status, error) {
+                        this.renderError("An error happened during request.", error);
+                    }.bind(this)
+                }), 
+                "#download-spinner", 
+                "download-spinner",
+            );
+        },
+        updateDataset: function(uuid, body, handlers){
+            if (overrides.updateDataset){
+                overrides.updateDataset(uuid, body, handlers, this);
+                return;
+            }
+
+            const { onSuccess = () => {}, onError = () => {} } = handlers;
+            $.ajax({
+                url: window.location.origin + "/picsure/dataset/named/" + uuid,
+                type: "PUT",
+                headers: { "Authorization": "Bearer " + JSON.parse(sessionStorage.getItem("session")).token },
+                contentType: "application/json",
+                dataType: "text",
```

Comment:
Why is this type text?

Reply:
If you change this to json, then ajax automatically parses to json, and throws an error when it can't. I'll change it to json and remove the manual parsing step.